pkg/helmreconciler/listeners.go

```diff
@@ -0,0 +1,472 @@
+package helmreconciler
+
+import (
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/helm/pkg/manifest"
+
+	"istio.io/operator/pkg/controller/common"
+)
+
+// CompositeRenderingListener is an implementation of RenderingListener which is composed of an array of listeners.
```

Comment:
better to separate these out into files.

Reply:
are you suggesting each type be separated out into individual files?

Reply:
this can be done in a subsequent pr, if necessary.